mongo/src/main/java/com/redhat/lightblue/mongo/crud/Translator.java

```diff
@@ -659,7 +691,31 @@ private DBObject translateRegexMatchExpression(RegexMatchExpression expr) {
         if (opStr.length() > 0) {
             regex.append("$options", opStr);
         }
-        return new BasicDBObject(translatePath(expr.getField()), regex);
+        return new BasicDBObject(translatePath(field), regex);
+    }
+
+    /**
+     * Get a reference to the path's hidden sub-field.
+     *
+     * This does not guarantee the sub-path exists.
+     *
+     * @param path
+     * @return
+     */
+    public static Path getHiddenForField(Path path) {
+        return path.prefix(-1).mutableCopy().push(HIDDEN_SUB_PATH).push(path.getLast());
+    }
```

Comment:
This isn't well defined if path itself is an instance of MutablePath, because then mutableCopy() returns this. Store path.getLast() in a local var first, then modify it.